packages/salesforcedx-vscode-core/src/conflict/conflictDetectionService.ts

```diff
@@ -0,0 +1,248 @@
+
+    // 1: create the shadow directory
+    try {
+      shell.mkdir('-p', projectMetadataTempPath);
+      shell.cp(data.manifest, retrievePackageXmlPath);
+    } catch (error) {
+      console.error(error);
+      channelService.appendLine(
+        nls.localize('error_creating_packagexml', error.toString())
+      );
+      notificationService.showErrorMessage(
+        nls.localize('error_creating_packagexml', error.toString())
+      );
+      return Promise.reject();
+    }
+
+    // 2: retrieve unmanaged org source to the shadow directory
+    await this.executeCommand(
+      this.buildRetrieveOrgSourceCommand(data),
+      { cwd: projectPath },
+      cancellationTokenSource,
+      cancellationToken
+    );
+
+    // 3: unzip retrieved source
+    try {
+      const zip = new AdmZip(unpackagedZipFile);
+      zip.extractAllTo(projectMetadataTempPath, true);
+    } catch (error) {
+      console.error(error);
+      channelService.appendLine(
+        nls.localize('error_extracting_org_source', error.toString())
+      );
+      notificationService.showErrorMessage(
+        nls.localize('error_extracting_org_source', error.toString())
+      );
+      return Promise.reject();
+    }
+
+    // 4: convert org source to decomposed (source) format
+    await this.executeCommand(
+      this.buildMetadataApiConvertOrgSourceCommand(data),
+      { cwd: projectPath },
+      cancellationTokenSource,
+      cancellationToken
+    );
+
+    // 5: diff project directory (local) and retrieved directory (remote)
+    // Assume there are consistent subdirs from each root i.e. 'main/default'
+    const diffs = this.differ.diff(localSourcePath, remoteSourcePath);
+
+    // 6: cleanup temp directory
+    try {
+      // shell.rm('-rf', projectMetadataTempPath);
+    } catch (error) {
+      // Failed cleanup should not terminate the service
+      console.error(error);
+      channelService.appendLine(
+        nls.localize('error_cleanup_temp_files', error.toString())
+      );
+      notificationService.showErrorMessage(
+        nls.localize('error_cleanup_temp_files', error.toString())
+      );
+    }
+
+    return diffs;
+  }
+
+  public logMetric(
+    logName: string | undefined,
+    executionTime: [number, number],
+    additionalData?: any
+  ) {
+    telemetryService.sendCommandEvent(logName, executionTime, additionalData);
+  }
+
+  public async executeCommand(
+    command: Command,
+    options: SpawnOptions,
+    cancellationTokenSource: vscode.CancellationTokenSource,
+    cancellationToken: vscode.CancellationToken
+  ): Promise<string> {
+    const startTime = process.hrtime();
+    const execution = new CliCommandExecutor(command, options, true).execute(
+      cancellationToken
+    );
+
+    const result = new CommandOutput().getCmdResult(execution);
+    this.attachExecution(execution, cancellationTokenSource);
+    execution.processExitSubject.subscribe(() => {
+      this.logMetric(execution.command.logName, startTime);
```

Comment:
You can directly call `telemetryService.sendCommandEvent` here and remove the `logMetric` method on line 208.